config/initializers/services.rb

```diff
@@ -0,0 +1,64 @@
+module SelfService
+
+  def self.register_service(name:, client:)
+    @services ||= {}
+
+    @services[name] = client
+  end
+
+  def self.service(name)
+    @services[name] || raise(ServiceNotRegisteredException.new(name))
+  end
+
+  class ServiceNotRegisteredException < RuntimeError; end
+end
+
+def aws_access_key
+  Rails.application.secrets.cognito_aws_access_key_id
+end
+
+def aws_secret_key
+  Rails.application.secrets.cognito_aws_secret_access_key
+end
+
+def stub_client
+  Rails.application.secrets.cognito_client_id = SecureRandom.uuid
```

Comment:
do we need to set this here if we're creating the stub client?

Reply:
yes...  it assumes that if you're creating a stub client that the environment variables aren't set in any meaningful way and this avoids a crash which can occur when this isn't set (like in test or dev)

Reply:
Strange, I wouldn't expect stubs to require the config...

Reply:
The stub expects to get the right config and errors if it gets a nil when it expects a value.  I think thats the right behaviour and is what I would expect.  Do some basic validation client side before even sending the request which is what we have here.